### PR TITLE
[GPU] Dump cuDNN GEMM fusion graphs to json files.

### PR DESCRIPTION
**Pull request by sergachev**
Requires cuDNN frontend v1.5: merge https://github.com/openxla/xla/pull/13757 first.

### CODE REVIEW

**Comment by sergachev** [comment]
Please review

File: xla/service/gpu/fusions/cudnn_test.cc [hunk 1]
```diff
@@ -83,6 +85,79 @@ class CuDnnFusionTest : public GpuCodegenTest {
...
+  TF_ASSERT_OK_AND_ASSIGN(bool changed, cudnn_compiler.Run(module.get()));
+  EXPECT_TRUE(changed);
+  std::string dump;
+  TF_EXPECT_OK(tsl::ReadFileToString(
+      tsl::Env::Default(),
+      tsl::io::JoinPath(output_directory,
+                        FilenameFor(*module, /*prefix=*/"",
+                                    /*suffix=*/"cudnn_fusion_d0.json")),
+      &dump));
+  EXPECT_TRUE(*RunFileCheck(dump, R"(
```
Comment (jreiffers) [inline xla/service/gpu/fusions/cudnn_test.cc:124]: Looks like filecheck doesn't do anything useful here. Maybe just EXPECT_EQ?

**Comment by jreiffers** [comment]
I was OOO and missed this. Looking now.

**Review by jreiffers — APPROVED** [review]

File: xla/service/gpu/fusions/cudnn_test.cc (hunk 1 above)
Comment (sergachev) [reply xla/service/gpu/fusions/cudnn_test.cc:124]: Checks below skip some lines with auto-generated properties

Comment (jreiffers) [reply xla/service/gpu/fusions/cudnn_test.cc:124]: OK, there's also testing::HasSubstr, which may be a bit nicer. But up to you.

**Pull request closed without merging**